src/main/java/org/apache/commons/dbcp2/BasicDataSource.java

```diff
@@ -1967,6 +1983,21 @@ public synchronized void setDriverClassName(final String driverClassName) {
             this.driverClassName = null;
         }
     }
+    
+    /**
+     * Sets the ConnectionFactory class name.
+     *
+     * @param connectionFactoryClassName
+     *            
+     */
+    
+    public void setConnectionFactoryClassName(final String connectionFactoryClassName) {
+    	if (connectionFactoryClassName != null && connectionFactoryClassName.trim().length() > 0) {

```

Comment:
`connectionFactoryClassName.trim().length() > 0` -> `!connectionFactoryClassName.trim().isEmpty()`

Reply:
Maybe even refactor into a private method `isNotBlank(String)` or `isBlank(String)` and reuse in other places in this class.